File: backend/src/main/java/com/bv87/diemdanh/dto/StaffRankDto.java
package com.bv87.diemdanh.dto;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class StaffRankDto {
    private final Integer rankCode;
    private final String rankCodeFormatted;
    private final String rankName;
    private final int sortOrder;
    private final boolean active;
    private final long usageCount;
}

File: desktop/src/BV87.Core/Models/Admin/Catalog/StaffRankDto.cs
namespace BV87.Core.Models.Admin.Catalog;

public sealed class StaffRankDto
{
    public int RankCode { get; set; }
    public string? RankCodeFormatted { get; set; }
    public string? RankName { get; set; }
    public int SortOrder { get; set; }
    public bool Active { get; set; }
    public long UsageCount { get; set; }
}

public sealed class StaffRankUpsertRequest
{
    public string RankName { get; set; } = string.Empty;
    public int? SortOrder { get; set; }
    public bool Active { get; set; } = true;
}
